### ISSUE TITLE
Global reuse of <list> <Entity> across intents  ??

### ISSUE THREAD

**Issue by KarlXOL** [microsoft/BotFramework-Composer#6244]
Either this is missing, not supported or couldn't figure out how to do this.

What I want to achieve is a bot with Luis, entity extraction providing some intents and using definitions for lists, entities across multiple intents. Imagine  a list like this for example:
@ list  carmodels =
    - car1
	-    car1
    - car2:
	-    car2
The same would be for entities with Luis.

When doing this definition in the intent it works as expected. 
But I want to avoid to reduplicate this definition in multiple intents.
I found no way to define this in a global lu and then to reuse it.
Any hints on that?

Thanks in advance

**Comment by feich-ms** [comment]
Hi @KarlXOL, 

1. If you want to reuse entities across intents of same dialog, you just need to define only once in any intent of that dialog since all the intent definitions under same dialog will be merged into one lu file per dialog and get published. Entity definition under one intent will be global reusable across different intents in that dialog, you don't need to define same entity in each intent.

2. If you want to reuse them across dialogs, you can manually define it in a lu file(you can name it freely), and import it to other lu files with import functionality. Suppose that we have a lu file defining the entities that will be reused across dialogs like below(we may need to manually create this file in the bot folder)

commom_entities.en-us.lu
```
@ list carmodels =
    - fuel:
        - car1, car2
    - electric:
        - car3, car4
```

If we want to reuse this entity definition in dialog A, we can add something like `[import common entities](./commom_entities.en-us.lu)` at top of dialog A lu file to import the entity into current lu file. If we want to reuse it in dialog B, we can do similar thing, import it in dialog B lu file. Please notice that you need to add the import command in the `Show code` view of dialog's lu file like below

![image](https://user-images.githubusercontent.com/43032123/110569102-00e13e00-818f-11eb-9a84-45b2d01d25ae.png)

**Comment by KarlXOL** [comment]
Thanks for feedback!
I tried option with importing luis to another .lu but when
doing 
[import](common.lg)(./commom_entities.en-us.lu)

I always get =>
Unexpected content: ‘import(commom_entities.en-us.lu)’. Expecting a comment, template definition, import statement or option definition.

Could you be a bit more specific on this solution? 
Thanks

**Comment by feich-ms** [comment]
@KarlXOL, I think you missed the square bracket, the format should be `[import](./commom_entities.en-us.lu)` and you can change the relative path in of `./commom_entities.en-us.lu` based on your file folder.

**Comment by KarlXOL** [comment]
I used this
![grafik](https://user-images.githubusercontent.com/21153915/110897290-78cd7680-82fd-11eb-972d-34e577ff4558.png)
get file not found. So what is the root?

thanks

**Comment by feich-ms** [comment]
@KarlXOL, we support robust relative path. You can create the common_entities.en-us.lu file under the folder YourBotRootFolder/language-understanding/en-us/ and then you can import them in each lu file just with `[import](common_entities.lu)`. For example, I have a ToDoBotWIthLuisSample-0 bot and I put the common_entities lu file under below folder
![image](https://user-images.githubusercontent.com/43032123/110899166-94577200-833b-11eb-9af7-1530ef71270b.png)

Then in lu files of each dialog, I just put the import at the header like below

![image](https://user-images.githubusercontent.com/43032123/110899377-eb5d4700-833b-11eb-9bcc-df99d364835f.png)

![image](https://user-images.githubusercontent.com/43032123/110899395-f31ceb80-833b-11eb-8954-087020682c7f.png)

When publishing the bot, the entities defined in common_entities.en-us.lu will be pulled to each dialog's luis applications like below

![image](https://user-images.githubusercontent.com/43032123/110899481-1fd10300-833c-11eb-82bf-8c07e3f3c0b9.png)

Hope this works.

**Comment by KarlXOL** [comment]
Thanks for detailed instructions. That was exactly what I tried.
Anyway, the import doesn't work as described.
Import in Dialog:
![grafik](https://user-images.githubusercontent.com/21153915/110905447-c7cdd880-830a-11eb-812b-88f700f52ca7.png)

lu-file in bot folder as described above
![grafik](https://user-images.githubusercontent.com/21153915/110905530-e7fd9780-830a-11eb-81de-dd2c4437ab85.png)

Interestingly the dialog tree shows the import then in the left
![grafik](https://user-images.githubusercontent.com/21153915/110905618-0b284700-830b-11eb-887f-609c515eb6d5.png)

**Comment by feich-ms** [comment]
@KarlXOL, if the common_entities.lu is in the same folder of you above lu file, you can use `./` to indicate the relative path, if it's in other folders, use the right relative path based on folder structure or just use [import](common_entities.lu) without `./`.  We recommend the latter one in that way our system will look for the lu file of that name globally in this bot folder. For the dialog tree, could you attach a more detailed screenshot? Which tab is it in?

**Comment by KarlXOL** [comment]
Shame on me. Probably I did changes not in the correct place!

Nevertheless your information provided details, very helpful to me and hopefully also for others

Many thanks!

**Comment by anusharavindrar** [comment]
@KarlXOL, is your issue resolved with the above comments from Fei? Shall I go ahead and close the issue?

**Comment by KarlXOL** [comment]
Thanks, I've tried this and it is working.
